src/windows.rs

```diff
@@ -127,12 +128,18 @@ impl Client {
         ))
     }
 
-    pub unsafe fn open(s: &str) -> io::Result<Client> {
-        let name = CString::new(s).map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+    pub unsafe fn open(s: &str, _check_pipe: bool) -> Result<Client, FromEnvError> {
+        let name = match CString::new(s) {
+            Ok(s) => s,
+            Err(e) => return FromEnvError::CannotParse(format!("not a valid CString: {e}")),
```

Comment:
```suggestion
            Err(e) => return FromEnvError::CannotParse(e.to_string())),
```
User doesn't care that we are using `CString` internally.